src/lotus_json/mod.rs

```diff
@@ -320,6 +320,40 @@ pub mod hexify_bytes {
     }
 }
 
+pub mod hexify_vec_bytes {
+    use super::*;
+    use std::fmt::Write;
+
+    pub fn serialize<S>(value: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = String::with_capacity(2 + value.len() * 2);
+        s.push_str("0x");
+        for b in value {
+            write!(s, "{:02x}", b).expect("failed to write to string");
+        }
+        serializer.serialize_str(&s)
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        #[allow(clippy::indexing_slicing)]
+        if s.len() >= 2 && &s[..2] == "0x" {
```

Comment:
You can avoid this warning with `.get(..2)`

Reply:
This avoid slice indexing but `split_at_checked` in unstable:
```rust
fn hex_str_to_bytes(value: &str) -> Result<Vec<u8>, anyhow::Error> {
    if let Some((head, tail)) = value.split_at_checked(2) {
        if head == "0x" && value.len() % 2 == 0 {
            return tail
                .as_bytes()
                .chunks(2)
                .map(|b| {
                    if let Ok(s) = std::str::from_utf8(b) {
                        u8::from_str_radix(s, 16).context("invalid digit found")
                    } else {
                        bail!("non-utf8 string")
                    }
                })
                .collect();
        }
    }
    bail!("invalid hex string")
}
```

Reply:
I don't really understand, how is it related?

Reply:
Never mind, it was a test to avoid using `unwrap/except`.
I'll go with `.get(..2)`.